### ISSUE TITLE
Error when visualizing the workflow dag

### ISSUE THREAD

**Issue by abonte** [snakemake/snakemake#2566]
**Snakemake version**

The error occurs in v8.0.1, and I've verified that it doesn't occur in v7.32.4

**Describe the bug**

Error when visualizing the workflow dag.

**Logs**

```
Traceback (most recent call last):
  File "/home/venv/lib/python3.11/site-packages/snakemake/cli.py", line 1934, in args_to_api
    dag_api.printdag()
  File "/home/venv/lib/python3.11/site-packages/snakemake/api.py", line 602, in printdag
    self.workflow_api._workflow.printdag()
  File "/homevenv/lib/python3.11/site-packages/snakemake/workflow.py", line 879, in printdag
    ignore_incomplete=self.execution_settings.ignore_incomplete,
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'ignore_incomplete'
``` 

**Minimal example**

1. Create  an Snakefile like the example below:

```
rule test_rule:
    output:
        "test.txt"
    shell:
        "touch test.txt"
```
2. As described in the [documentation](https://snakemake.readthedocs.io/en/stable/executing/cli.html#visualization), run `snakemake --dag | dot | display`or `snakemake --dag | dot -Tpdf > dag.pdf`

**Additional context**

Python 3.11.6

**Comment by KasperSkytte** [comment]
I get this error too using `8.0.1`. Very simple workflow

**Comment by KasperSkytte** [comment]
Same with `8.1.0`, identical workflow

**Comment by abonte** [comment]
Same issue #2577

**Comment by Nicolas-Fernandez** [comment]
Hi dear all,
Same in 8.0.0 to 8.2.0. I need to revert to 7.xxx
Thank for your help, have a nice day,
Best,
Nicolas

**Comment by mhorlacher** [comment]
Any solutions?

**Comment by abonte** [comment]
Similar error #2637

**Comment by abonte** [comment]
Solved in v8.4.0

**Comment by abonte** [comment]
Solved in v8.4.0
